OpenRiaServices.DomainServices.Client/Test/Desktop/Data/DomainClientTests.cs

```diff
@@ -59,39 +60,19 @@ public void TestMockClient_Query()
             EnqueueTestComplete();
         }
         [TestMethod]
-        [Asynchronous]
         public void TestMockClient_CancellationSupport()
         {
             Cities.CityDomainContext dp = new Cities.CityDomainContext(new CitiesMockDomainClient());
 
-            bool domainClientQueryCompleted = false;
-            dp.DomainClient.BeginQuery(
-                new EntityQuery<Cities.City>(dp.DomainClient, "GetCities", null, true, false),
-                delegate(IAsyncResult getCitiesAsyncResult)
-                {
-                    ExceptionHelper.ExpectException<NotSupportedException>(delegate
-                    {
-                        dp.DomainClient.CancelQuery(getCitiesAsyncResult);
-                    }, string.Format(CultureInfo.CurrentCulture, Resource.DomainClient_CancellationNotSupported, typeof(CitiesMockDomainClient).FullName));
-
-                    domainClientQueryCompleted = true;
-                }, null);
+            var query = dp.GetCitiesQuery();
+            LoadOperation lo = dp.Load(query, false);
+            Assert.IsFalse(lo.CanCancel, "Cancellation should not be supported.");
+            Assert.IsFalse(dp.DomainClient.SupportsCancellation, "Cancellation should not be supported.");
 
-            EnqueueConditional(() => domainClientQueryCompleted);
-            EnqueueCallback(delegate
+            ExceptionHelper.ExpectException<NotSupportedException>(delegate
```

Comment:
double check the changes in this test, original test trid to cancel after completion